### PR TITLE
Add support for tolerations

### PR DESCRIPTION
**Pull request by samsonov90**
Added support for specifying tolerations for driver/executor pods that allows scheduling on nodes with taints.

### CODE REVIEW

**Comment by samsonov90** [comment]
@liyinan926 please take a look. We need to run Spark driver & executor pods on nodes with taints. And this PR is solving the problem.

File: pkg/config/toleration.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	for key, value := range annotations {
+		options = append(options, GetExecutorAnnotationOption(key, value))
+	}
+
+	return options, nil
+}
+
+// FindTolerations finds and parses Tolerations in the given annotations.
+func FindTolerations(annotations map[string]string) (map[string]*apiv1.Toleration, error) {
+
```
Comment (liyinan926) [inline pkg/config/toleration.go]: Additional empty line.

Comment (liyinan926) [reply pkg/config/toleration.go]: Also looks like you only need the `Toleration` objects themselves, so you can return a `[]*apiv1.Toleration` instead.

Comment (samsonov90) [reply pkg/config/toleration.go]: @liyinan926 addressed your comments

**Pull request merged**